src/client.rs

```diff
@@ -410,7 +448,7 @@ where
 
         // The query router determines where the query is going to go,
         // e.g. primary, replica, which shard.
-        let mut query_router = QueryRouter::new();
+        let mut query_router = QueryRouter::new(self.target_pool.clone());
```

Comment:
Query routing configs are per-pool now and not global so the router needs access to the pool configs